src/Common/tests/System/Diagnostics/RemoteExecutorTestBase.cs

```diff
@@ -4,17 +4,18 @@
 
 using System.IO;
 using System.Reflection;
+using System.Runtime.InteropServices;
 using Xunit;
 
 namespace System.Diagnostics
 {
     /// <summary>Base class used for all tests that need to spawn a remote process.</summary>
     public abstract class RemoteExecutorTestBase : FileCleanupTestBase
     {
-        /// <summary>The CoreCLR host used to host the test console app.</summary>
-        protected const string HostRunner = "corerun";
         /// <summary>The name of the test console app.</summary>
         protected const string TestConsoleApp = "RemoteExecutorConsoleApp.exe";
+        /// <summary>The CoreCLR host used to host the test console app.</summary>
+        protected static readonly string HostRunner = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "CoreRun.exe" : "corerun";
```

Comment:
Interesting I didn't catch that below. Off the top of my head I don't have a better solution to that, I mainly asked because I was going to see if we cannot just use the lowercase corerun in all the places to avoid this fork in logic that is duplicated everywhere. This is one case where I see there is some reason for it but perhaps in other cases there still isn't any real reason for it.